src/app/services/post.service.ts

```diff
@@ -29,14 +28,14 @@ export class PostService {
   //       Some external observer must subscribe to trigger the call.
   //       The `async` pipe performs the `.subscribe()`/`.unsubscribe()`
 
-  posts$ = this.http.get(this.allPostsUrl)
-      .map(res => res.json())
-      .mergeMap(posts=>{
-        return Observable.of(posts)
-            .merge(Observable.forkJoin(posts
-                .map(post=>this.user.getUser(post.userId)
-                    .map(user=>({...post,  authorName:user.name })))));
-      });
+  readonly posts$: Observable<PostWithAuthor[]> = this.http.get(this.allPostsUrl)
+    .flatMap(res => res.json() as Post[])
```

Comment:
Why are you using `flatMap` the emitted values are JSON objects not `Observables<JSON>`?

Reply:
No, `flatMap` is alias of `mergeMap`, they both emit `Observable<Post[]>`, being used here as a notion of flattening to the downstream.